src/hotspot/share/code/compiledMethod.cpp

```diff
@@ -453,6 +453,19 @@ void CompiledMethod::clear_ic_callsites() {
   }
 }
 
+void CompiledMethod::clear_continuation_enter_special_inline_caches() {
+  assert (method()->is_continuation_enter_intrinsic(), "not Continuation.enterSpecial");
+
+  ResourceMark rm;
+  CompiledICLocker ml(this);
+  RelocIterator iter(this);
+  while(iter.next()) {
+    if (iter.type() == relocInfo::static_call_type) {
+      iter.reloc()->clear_inline_cache();
```

Comment:
This relies on code patching, and for correctness the change must be seen by the thread requesting interpreter-only mode.  If this was being done at a safepoint then it would probably be OK.  However, this code seems to be done using a handshake, so I'm not sure if the required serializing instruction is guaranteed to happen (see JDK-8220351).

Reply:
The change is not guaranteed to be seen on a concurrent thread, until the next global handshake operation completes.

Reply:
If that concurrent thread is in interp_only_mode, it also would have done the same patching. And if it isn't, then it's okay for it not to see this, but if it does see it, it will re-patch to compiled in c2i, as in the description.